agent/internal/mapper/grpc.go

```diff
@@ -70,42 +71,180 @@ func MapDeployImage(req *agent.DeployRequest, appConfig *config.CommonConfigurat
 	return res
 }
 
-func MapContainerConfig(in *agent.DeployRequest_ContainerConfig) v1.ContainerConfig {
+func MapContainerConfig(in *agent.DeployRequest) v1.ContainerConfig {
 	containerConfig := v1.ContainerConfig{
-		Container: in.Name,
-		Ports:     MapPorts(in.Ports),
-		User:      &in.User,
+		Container:        in.Name,
+		ContainerPreName: in.InstanceConfig.Prefix,
+		Ports:            MapPorts(in.ContainerConfig.Ports),
+		PortRanges:       MapPortRanges(in.ContainerConfig.PortRanges),
+		Volumes:          MapVolumes(in.ContainerConfig.Volumes),
```

Comment:
Since we are using these mapper functions only here, it would be nice to make these private.